document ParentClosePolicy (#461)
temporalio/documentation: docs/go/workflows.md

we.Get(ctx, &result)
```

### ParentClosePolicy

> When creating a child Workflow, you can define a `ParentClosePolicy` that terminates, cancels, or abandons the Workflow if the child's parent stops execution.

Reviewer: ```suggestion
When creating a child Workflow, you can define a `ParentClosePolicy` that terminates, cancels, or abandons the Workflow Execution if the child's parent stops execution.
```

> - `ABANDON`: When the parent stops, don't do anything with the child workflow.
> - `TERMINATE`: When the parent stops, terminate the child workflow
> - `REQUEST_CANCEL`: When the parent stops, terminate the child workflow

Reviewer: ```suggestion
- `ABANDON`: When the parent stops, don't do anything with the Child Workflow.
- `TERMINATE`: When the parent stops, terminate the Child Workflow
- `REQUEST_CANCEL`: When the parent stops, terminate the Child Workflow
```

You can set policies per child, which means you can opt out of propagating terminates / cancels on a per-child basis.
This is useful for starting Child Workflows asynchronously:

> 1. Set `ChildWorkflowOptions.ParentClosePolicy` to `ABANDON` when creating a child workflow stub.

Author: no - oversight on my part

> 2. Start child workflow asynchronously using `ExecuteChildWorkflow`.

Reviewer: ```suggestion
2. Start the Child Workflow Execution asynchronously using `ExecuteChildWorkflow`.
```

3. Call `GetChildWorkflowExecution` on the `ChildWorkflowFuture` returned by the `ChildWorkflowFuture`

> 4. Wait for the `ChildWorkflowFuture`. This indicates that the child successfully started (or start failed).

Reviewer: ```suggestion
4. Wait for the `ChildWorkflowFuture`.
    This indicates that the child successfully started (or start failed).
```

> 5. Complete parent workflow asynchronously.

Reviewer: ```suggestion
5. Complete Parent Workflow Execution asynchronously.
```

> Steps 3 and 4 are needed to ensure that child workflow starts before the parent closes. If the parent initiates child workflow and immediately completes the child would never start.

Reviewer: ```suggestion
Steps 3 and 4 are needed to ensure that a Child Workflow Execution starts before the parent closes. 
    If the parent initiates a Child Workflow Execution and then immediately completes, the child would never execute.
```
